client/Components/Items/AllItems.js

```diff
@@ -1,19 +1,39 @@
-import React from "react";
+import React, { Component } from "react";
 import { connect } from "react-redux";
 import ItemPreview from "./ItemPreview";
+import { itemsFetchStartAsync } from "../../Redux/Items/actions/items.actions";
 
-export const AllItems = ({ inventory }) => {
-  const generateItemCards = inventoryArr => {
-    if (inventoryArr.length > 0) {
-      return inventoryArr.map(item => <ItemPreview item={item} />);
-    }
-    return <div>Items are loading...</div>;
-  };
+export class AllItems extends React.Component {
+  componentDidMount() {
+    const { itemsFetchStartAsync } = this.props;
+    itemsFetchStartAsync();
+  }
+  render() {
+    const { inventory } = this.props;
+    const generateItemCards = inventoryArr => {
```

Comment:
Should keep functions/maps outside of the render or final returns.

Reply:
Absolutely, works for me.